DataProc/skimming/skimRawCENTEJ1.C

```diff
@@ -0,0 +1,377 @@
+  std::cout << "mask   " << std::bitset<64>(mask) << "  ("<<mask<<")"<<endl;
+  std::cout << "mask50 " << std::bitset<64>(maskNext50) << "  ("<<maskNext50<<")"<< endl;
+  printf("--------------------------------------------------------------------------\n");
+  //////////////////
+  
+  //  return Process(Form("%s/000%d/raw/", period.Data(),runNb), chunkName, mask, maskNext50)) {
+  bool res = Process(chain,  mask, maskNext50);
+  wc.Stop();
+  wc.Print();
+  printf("Bytes: read: %ld written %ld\n",bread,bwrite);
+  return res;
+}
+
+
+const char *GetRootAliRootInfo()
+{
+  // Return Root/AliRoot version and revisiosn 
+
+  static TString version;
+  version = "AliRoot version=";
+  version+=TString(ALIROOT_VERSION);
+  version+=", reversion=";
+  version+=TString(ALIROOT_REVISION);
+  version+= "; Root version=";
+  version+=gROOT->GetVersion();
+  return version.Data();
+}
+
+
+static void BranchResetBit(TBranch *b) 
+{
+  // Reset MapObject on this branch and all the sub-branches
+
+  b->ResetBit( kBranchObject | kBranchAny ); // Or in newer ROOT: b->ResetBit( kMapObject )
+  TIter next( b->GetListOfBranches() );
+  TBranch *sub = 0;
+  while ( (sub = (TBranch*)next() ) ) {
+    BranchResetBit(sub);
+  }
+}
+
+
+Bool_t Process(TChain *rawTree, const ULong64_t mask_1, const ULong64_t maskNext50_1)
+{
+  int run = AliCDBManager::Instance()->GetRun();
+  TFile newfile(Form("skim%09d.root",run),"recreate");
+    
+  TTree *newtree = new TTree("RAW", Form("ALICE skimmed raw-data tree (%s)", GetRootAliRootInfo()));
+  Int_t basketsize = 32000;
+  Long64_t autoflush=-5000000LL;
+  Int_t split   = 99;
+  newtree->SetAutoSave(21000000000LL);  // autosave when 21 Gbyte written
+  newtree->SetAutoFlush(autoflush);
+  newtree->BranchRef();
+  
+  AliRawEventV2   *rawEvent=NULL; 
+  TObjArray brNames; // names of branches to read
+  brNames.SetOwner(kTRUE); 
+
+  TNamed* bname = new TNamed("rawevent","");
+  brNames.Add( bname ); // this must be the 1st branch to read
+
+  rawTree->SetBranchAddress( bname->GetName() , &rawEvent);
+  TBranch *b = newtree->Branch( bname->GetName() , "AliRawEventV2", &rawEvent, basketsize, split);
+  BranchResetBit(b);
+
+  Int_t detBranches[AliDAQ::kNDetectors+1] = {
+    1, /*  1 ITSSPD */
+    1, /*  2 ITSSDD */
+    1, /*  3 ITSSSD */
+    18, /*  4 TPC */
+    1, /*  5 TRD */
+    1, /*  6 TOF */
+    1, /*  7 HMPID */
+    1, /*  8 PHOS */
+    1, /*  9 CPV */
+    1, /* 10 PMD */
+    1, /* 11 MUONTRK */
+    1, /* 12 MUONTRG */
+    1, /* 13 FMD */
+    1, /* 14 T0 */
+    1, /* 15 VZERO */
+    1, /* 16 ZDC */
+    1, /* 17 ACORDE */
+    1, /* 18 TRG */
+    1, /* 19 EMCAL */
+    1, /* 20 DAQ_TEST */
+    1, /* 21 EMPTY */
+    1, /* 22 AD */
+    1, /* 23 MFT */
+    1, /* 24 FIT */
+    10,/* 25 HLT */
+    1  /* 26 --- */
+  };
+
+
+  // select branches	
+  Bool_t kDetAccept[AliDAQ::kNDetectors+1] = {
+    kTRUE , //SPD
+    kFALSE, //SDD
+    kTRUE , //SSD
+    kTRUE,  //TPC
```

Comment:
@mfasDa we were running in HLT-C mode, therefore the TPC branches will be empty (still ok to have it).